mitmproxy/tools/web/app.py

```diff
@@ -324,6 +327,72 @@ def broadcast(cls, **kwargs):
 class ClientConnection(WebSocketEventBroadcaster):
     connections: ClassVar[set] = set()
 
+    def __init__(self, application: Application, request, **kwargs):
+        super().__init__(application, request, **kwargs)
+        self.application: Application = application
+        self.filters: dict[str, flowfilter.TFilter] = {}  # Instance-level filters
+
+    def send_flow(self, cmd: str, f: mitmproxy.flow.Flow):
+        matches = [
+            {expr.pattern: bool(flowfilter.match(expr, f))}
+            for expr in self.filters.values()
+        ]
```

Comment:
```suggestion
        matches = {
            expr.pattern: expr(f)
            for expr in self.filters.values()
        }
```

If `bool()` is necessary, then the type annotation for TFilter is incorrect and we should fix that.

Reply:
without an explicit cast to bool, this doesn't work for some reasons

Reply:
Addressed in [3ff2680](https://github.com/mitmproxy/mitmproxy/pull/7506/commits/3ff26802302666afef497cc26fad46640af55bf5).

Reply:
thanks :)